rs/registry/canister/src/registry.rs

```diff
@@ -180,9 +184,22 @@ impl Registry {
     /// Increments the latest version of the registry.
     fn increment_version(&mut self) -> Version {
         self.version += 1;
+        self.record_version_at_timestamp();
```

Comment:
TODO write a test that the map is filled out correctly and always writes the entries, and gets the timestamp increments right.